lib/libzrepl/data_conn.c

```diff
@@ -1213,6 +1245,16 @@ uzfs_zvol_rebuild_scanner(void *arg)
 			    == 1)
 				sleep(5);
 #endif
+
+			snap_zv = uzfs_get_snap_zv_ionum(zinfo,
+			    hdr.checkpointed_io_seq);
+			if ((snap_zv == NULL) && (all_snap_done == B_FALSE)) {
```

Comment:
all_snap_done should be false if snap_zv is null.. so, an assert can be added, or, can be errored out due to error in protocol

Reply:
done